### PR TITLE
Use custom iree.dev domain in links to documentation site.

### PR DESCRIPTION
**Pull request by ScottTodd**
Follow-up to https://github.com/openxla/iree/pull/15008 and some GitHub configuration changes that switched us to using a custom domain for our GitHub pages hosting.

Background:

* We acquired this domain for hosting https://perf.iree.dev/ years ago, with some aspirations to use the domain for other pages eventually.
* We tried to use the domain with our GitHub Pages content at one point, but ran into some issues that resulted in buggy redirects and certificate errors. A few weeks ago we tried that again and succeeded.

Note that we had also considered using the `docs` subdomain, i.e. `https://docs.iree.dev/`. That still seems tempting, but would require more domain configuration work. Until we use a subdomain like that or explore the workarounds in https://github.com/orgs/community/discussions/23184, the `www` part of URLs is required.

### CODE REVIEW

**Comment by jpienaar** [comment]
Shall we try to see if docs works before doing this? (Up to you)

**Comment by ScottTodd** [comment]
> Shall we try to see if docs works before doing this? (Up to you)

Sure.

**Review by antiagainst — APPROVED** [review]
Nice. LGTM whenever you are ready. :)

**Comment by ScottTodd** [comment]
> > Shall we try to see if docs works before doing this? (Up to you)
> 
> Sure.

Were you going to try setting up `docs.iree.dev` alongside / instead of `www.iree.dev`, @jpienaar ? If so, what's your timeline like? (Also, can I help?)

File: docs/website/mkdocs.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 site_name: IREE
-site_url: https://openxla.github.io/iree/
+site_url: https://www.iree.dev/
```
Comment (ScottTodd) [inline docs/website/mkdocs.yml]: Hmm... looks like the 404 page is broken with the current URL. Updating this site url may help.

https://www.iree.dev/404
```
 GET https://www.iree.dev/iree/assets/stylesheets/main.0e669242.min.css net::ERR_ABORTED 404 (Not Found)
 GET https://www.iree.dev/iree/assets/stylesheets/palette.85d0ee34.min.css net::ERR_ABORTED 404 (Not Found)
 GET https://www.iree.dev/iree/assets/stylesheets/extra.css net::ERR_ABORTED 404 (Not Found)
 GET https://www.iree.dev/iree/assets/stylesheets/openxla.css net::ERR_ABORTED 404 (Not Found)
 GET https://www.iree.dev/iree/assets/javascripts/bundle.78eede0e.min.js net::ERR_ABORTED 404 (Not Found)
 GET https://www.iree.dev/iree/assets/images/openxla-logo-lockup-white.svg 404 (Not Found)
```
Note the extra `iree/` in those paths. The URLs in the .html are of the form `<link rel="stylesheet" href="/iree/assets/stylesheets/extra.css">`

**Comment by jpienaar** [comment]
> > > Shall we try to see if docs works before doing this? (Up to you)
> > 
> > 
> > Sure.
> 
> Were you going to try setting up `docs.iree.dev` alongside / instead of `www.iree.dev`, @jpienaar ? If so, what's your timeline like? (Also, can I help?)

Sorry missed that you had approved the other change.

**Comment by jpienaar** [comment]
Submitted and will see if that works or if I need another update.

**Comment by ScottTodd** [comment]
> Submitted and will see if that works or if I need another update.

Doesn't seem to be working on its own? I changed the custom domain in the repository settings:

image

Here's with www (which I just switched back to):

image

**Comment by ScottTodd** [comment]
Rebased and dropped the `www.` prefix.

File: docs/website/mkdocs.yml (hunk 1 above)
Comment (ScottTodd) [reply docs/website/mkdocs.yml]: Nice, this fixed the 404 page (e.g. https://iree.dev/404)

**Pull request merged**